### PR TITLE
Support ok http

### PR DESCRIPTION
**Pull request by khalil2535**
- Replace Volley with OkHttp
- Upload application icon

### CODE REVIEW

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        Request request = new Request.Builder().url(uri).post(body).build();
+
+        return client.newCall(request).execute();
+    }
+
+    static Response makeFormPostRequest(String uri, String[][] dataLines, String action)
+            throws IOException {
+
+        Log.d(TAG, "making a Form Post Request");
```
Comment (WaleedMortaja) [inline app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java]: Log the parameters values too.

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        Request request = new Request.Builder().url(uri).post(body).build();
+
+        return client.newCall(request).execute();
+    }
+
+    static Response makeFormPostRequest(String uri, String[][] dataLines, String action,
+                                        String format) throws IOException {
+
+        Log.d(TAG, "making a Form Post Request");
```
Comment (WaleedMortaja) [inline app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java]: Log the parameters values too.

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        } else {
+            Log.e(TAG, "CheckLoginAuthorization" + (
+                    response.trim().isEmpty() ? "Empty Response" : "Unknown Authorization"));
+
+            return Authorization.UNKNOWN;
+        }
+    }
+
+    private static Authorization getAuthorization(String username, String password) {
+        final String login = "login";
```
Comment (WaleedMortaja) [inline app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java]: - I think this variable should be named ACTION_LOGIN
- I wonder what other acceptable actions could be ?

File: app/src/main/java/ps/edu/ucas/ucashub/activities/LoginActivity.java [hunk 4]
```diff
@@ -35,9 +41,19 @@
...
     @Click(R.id.lgoinBtn)
     public void login(View view) {
-        // TODO establishNetworkConnection then login in moodle & student page
+        if (MainWebServices.establishNetworkConnection()) {
+            String username = usernameEtxt.getText().toString();
+            String password = passwordEtxt.getText().toString();
+            if (MoodleServices.Login(username, password))
+                Log.d(TAG, "Done");
+            else
+                Log.d(TAG, "can't connect to moodle");
```
[the remark this answers is not in the document]
Comment (khalil2535) [reply app/src/main/java/ps/edu/ucas/ucashub/activities/LoginActivity.java]: Ok.

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/MainWebServices.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                Log.e(TAG, "Cannot establish connection to Google. Exception : " + e.toString());
+                googleConnected = false;
+            }
+        }
+
+        if (!UCASPageNetworkConnected) {
+            try {
+                Response response = Requests.makeGetRequest(UCAS_URL);
+                UCASPageNetworkConnected = response.isSuccessful();
+                Log.d(TAG, "connected to ucas");
```
[the remark this answers is not in the document]
Comment (khalil2535) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/MainWebServices.java]: Ok done.

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java (hunk 1 above)
Comment (khalil2535) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java]: do i have to log all the parameters for all the methods or you mean just this method you specified ?

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java (hunk 2 above)
Comment (khalil2535) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java]: Ok Done.

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java (hunk 3 above)
Comment (khalil2535) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java]: - [x] change variable name.

about the second question as you know the forms are not just for login , we will use forms again in our app to update the moodle profile for example. (i don't know the action until now).

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java (hunk 1 above)
Comment (WaleedMortaja) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/Requests.java]: All the methods, so when an exception or any problem accures. You would know the parameters that cause the problem. <b>Do NOT reveal the password</b>

File: app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java (hunk 3 above)
Comment (WaleedMortaja) [reply app/src/main/java/ps/edu/ucas/ucashub/services/web/MoodleServices.java]: Okay

**Pull request merged**